src/benchmarks/micro/ValuesGenerator.cs

```diff
@@ -21,6 +22,12 @@ public static T GetNonDefaultValue<T>()
                 return ArrayOfUniqueValues<T>(2).First(value => !value.Equals(default));
         }
 
+        public static T GetValue<T>(T minValue, T maxValue)
+        {
+            var random = new Random(Seed);
```

Comment:
this is very tricky: this method will always return the same result because we use a const seed.

```cs
for (int i = 0; i < 10; i++)
    Console.WriteLine(ValuesGenerator.GetValue(1.0, double.PositiveInfinity));
```

```
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
3,98113421984948E+20
```

In general, this is the desired approach. However, in our case, we want to have a different number per iteration to make sure the JIT does not cache the result?

Reply:
Ah, right. We would need to have it keep the same random that was initially seeded, which would also require the tests to run in a deterministic order.

Reply:
@gfoidl we should rather create it inside a method and pass to other methods like here:

https://github.com/dotnet/performance/blob/c4efbbb55e0a5d8f4408c9806bdb51b300a261fc/src/benchmarks/micro/ValuesGenerator.cs#L29-L35

so anytime the method is invoked it produces the same results

Reply:
> JIT does not cache the result?

By what mechanism would the JIT do that? It doesn't know enough to know it's the same result each time does it?

Reply:
A number of the `System.Math` functions are intrinsicly recognized by the JIT and will under go constant folding since they should be deterministic.